tests/phpunit/unit-tests/user/class-llms-test-user-permissions.php

```diff
@@ -164,6 +164,40 @@ public function test_user_can_manage_user_multiple_roles() {
 
 	}
 
+	/**
+	 * Test the editable_roles() filter for users with multiple roles.
+	 *
+	 * @since [version]
+	 *
+	 * @return void
+	 */
+	public function test_editable_roles_multiple_roles() {
+
+		$users = $this->create_mock_users();
+
+		$all_roles = LLMS_Roles::get_roles();
+
+		wp_set_current_user( $users['lms_manager'] );
+		$lms_manager_editable_roles = LLMS_Unit_Test_Util::call_method( $this->obj, 'editable_roles', array( $all_roles ) );
+
+		wp_set_current_user( $users['instructor'] );
+		$instructor_editable_roles = LLMS_Unit_Test_Util::call_method( $this->obj, 'editable_roles', array( $all_roles ) );
+
+		wp_set_current_user( $users['lms_manager'] );
+		$user = wp_get_current_user();
+		$user->add_role( 'instructor' );
+		$editable_roles = LLMS_Unit_Test_Util::call_method( $this->obj, 'editable_roles', array( $all_roles ) );
+
+		// assert that lms_manager with instructor role has editable roles from both roles
+		foreach ( $lms_manager_editable_roles as $lms_manager_editable_role ) {
+			$this->assertContains( $lms_manager_editable_role, $editable_roles );
+		}
+		foreach ( $instructor_editable_roles as $instructor_editable_role ) {
+			$this->assertContains( $instructor_editable_role, $editable_roles );
+		}
+
```

Comment:
This implementation works for any 2 roles, except `administrator`, because `administrator` is not configured in `get_editable_roles()`. I think the best way might be to specifically check for `administrator`, because I don't think we want to list all the roles the `administrator` can edit in `get_editable_roles()`, or it would require updating that method everytime a new role is added

Reply:
I've catered for the `administrator` role, and I've added regression tests